alcs-api/src/application/application-decision/application-decision.controller.ts

```diff
@@ -0,0 +1,190 @@
+import { Mapper } from '@automapper/core';
+import { InjectMapper } from '@automapper/nestjs';
+import {
+  BadRequestException,
+  Body,
+  Controller,
+  Delete,
+  Get,
+  NotFoundException,
+  Param,
+  Patch,
+  Post,
+  Req,
+  UseGuards,
+} from '@nestjs/common';
+import { ApiOAuth2 } from '@nestjs/swagger';
+import * as config from 'config';
+import { RoleGuard } from '../../common/authorization/role.guard';
+import { ANY_AUTH_ROLE } from '../../common/authorization/roles';
+import { UserRoles } from '../../common/authorization/roles.decorator';
+import { ApplicationService } from '../application.service';
+import { ApplicationDecisionOutcome } from './application-decision-outcome.entity';
+import {
+  ApplicationDecisionDto,
+  ApplicationDecisionOutcomeDto,
+  CreateApplicationDecisionDto,
+  DecisionDocumentDto,
+  UpdateApplicationDecisionDto,
+} from './application-decision.dto';
+import { ApplicationDecision } from './application-decision.entity';
+import { ApplicationDecisionService } from './application-decision.service';
+import { DecisionDocument } from './decision-document.entity';
+
+@ApiOAuth2(config.get<string[]>('KEYCLOAK.SCOPES'))
+@Controller('application-decision')
+@UseGuards(RoleGuard)
+export class ApplicationDecisionController {
+  constructor(
+    private appDecisionService: ApplicationDecisionService,
+    private applicationService: ApplicationService,
+    @InjectMapper() private mapper: Mapper,
+  ) {}
+
+  @Get('/application/:fileNumber')
+  @UserRoles(...ANY_AUTH_ROLE)
+  async getAllForApplication(@Param('fileNumber') fileNumber): Promise<{
+    decisions: ApplicationDecisionDto[];
+    codes: ApplicationDecisionOutcomeDto[];
+  }> {
+    const decisions = await this.appDecisionService.getByAppFileNumber(
+      fileNumber,
+    );
+    const codes = await this.appDecisionService.getCodeMapping();
+    const mappedDecisions = this.mapper.mapArray(
+      decisions,
+      ApplicationDecision,
+      ApplicationDecisionDto,
+    );
+    const mappedCodes = this.mapper.mapArray(
+      codes,
+      ApplicationDecisionOutcome,
+      ApplicationDecisionOutcomeDto,
+    );
+
+    return {
+      decisions: mappedDecisions,
+      codes: mappedCodes,
```

Comment:
why not include code into CodeController?

Reply:
Its only needed on a single page, I figure this was better than loading it all the time even when its not needed.